src/test/java/org/fungover/haze/MainTest.java

```diff
@@ -1,19 +1,50 @@
 package org.fungover.haze;
-
+import org.apache.logging.log4j.core.jmx.Server;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.CsvSource;
 import org.junit.jupiter.params.provider.ValueSource;
-
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
-
 import static org.assertj.core.api.Assertions.assertThat;
-
+import static org.assertj.core.api.Assertions.in;
+import static org.mockito.Mockito.*;
 
 class MainTest {
+
+    private ServerSocket serverSocket;
+    private Socket clientSocket;
+
+    @BeforeEach
+    void setUp() throws IOException {
+        serverSocket = new ServerSocket(0);
```

Comment:
Might be a bad idea since we are not really testing our code more testing ServerSocket functions.